Fix `String::rfindn` for strings with only one character.

Yarwin opened the pull request:
Our test for rfindn (https://github.com/godot-rust/gdext/blob/5e75d1c7f76762e754ffa9c9c17218068ebb1be0/itest/rust/src/builtin_tests/string/gstring_test.rs#L164) started failing while searching for one-character long sequences (i.e. `"Hello World".rfindn("O")` returns -1 instead of 7)

I was a bit confused where new function should go (do lowercase spans even make sense???), so I just monkey'd implementation of `strings_equal_lower`.

- _Bugsquad edit: Fixes regression from https://github.com/godotengine/godot/pull/101247_

AThousandShips commented on core/string/ustring.h:
```diff
@@ -31,7 +31,6 @@
 #pragma once
 
 // Note: _GODOT suffix added to header guard to avoid conflict with ICU header.
-
```
Unrelated and will cause a rebuild of the entire engine, please restore

Ivorforce commented on core/string/ustring.cpp:
```diff
@@ -3213,7 +3225,7 @@ int String::rfindn(const String &p_str, int p_from) const {
 		// Optimize with single-char implementation.
-		return span().rfind(p_str[0], p_from);
+		return span_rfind_lower(span(), p_str[0], p_from);
 	}
```
```suggestion
```

Ivorforce reviewed (commented):
Thanks for following up so fast!

Most of the functions that have a single-char optimization use it because the single-char function exists already for something else.
Since that isn't the case here (i.e. no `rfindn_char`), I'd just remove the optimization instead. That saves us the complexity of adding a function.

Yarwin committed:
Fix String::rfindn for strings with only one character.

----

Remove optimization – no need to introduce extra complexity for
negligible (if any) gains.
